src/test/ui/rfcs/rfc-2396-target_feature-11/fn-traits.stderr

```diff
@@ -0,0 +1,39 @@
+error[E0277]: expected a `std::ops::Fn<()>` closure, found `fn() {foo}`
```

Comment:
I find this error is confusing and misleading, but at the same time it is consistent with error emitted for unsafe functions. Maybe we want to change it to better explain why foo doesn’t implement `Fn<()>`?

Reply:
I added a note (open to phrasing suggestions), however I didn't remove this potentially misleading note since it's applied to all closures without any arguments.  I'm not sure if it's within the scope of this PR to fix the `#[rustc_on_unimplemented]` on the Fn traits, but it's misleading for `unsafe fn()` too, I think.